### PR TITLE
[wgsl-in] consolidate type resolution logic in a few macros

### PR DESCRIPTION
**Pull request by teoxoy**

### CODE REVIEW

File: src/front/wgsl/lower/mod.rs [hunk 1]
```diff
@@ -13,6 +13,32 @@ use crate::{Arena, FastHashMap, FastIndexMap, Handle, Span};
...
+pub(super) use resolve;
+
+macro_rules! resolve_inner {
+    ($ctx:ident, $expr:expr) => {{
+        $ctx.grow_types($expr)?;
+        $ctx.typifier()[$expr].inner_with(&$ctx.module.types)
+    }};
+}
+
+macro_rules! resolve_inner_x2 {
```
Comment (jimblandy) [inline src/front/wgsl/lower/mod.rs:31]: Suggest name: `resolve_inner_binary_op` or `resolve_inner_binary`.

File: src/front/wgsl/lower/mod.rs [hunk 2]
```diff
@@ -1146,11 +1162,7 @@ impl<'source, 'temp> Lowerer<'source, 'temp> {
 
                     let initializer = match v.init {
                         Some(init) => {
-                            let initializer =
-                                self.expression(init, ctx.as_expression(block, &mut emitter))?;
-                            ctx.as_expression(block, &mut emitter)
-                                .grow_types(initializer)?;
```
Comment (jimblandy) [inline src/front/wgsl/lower/mod.rs:1152]: Was this call just completely extraneous?

**Review by jimblandy — APPROVED** [review]
This is a nice cleanup.

These need documentation, both for usage and their rationale for existence.

File: src/front/wgsl/lower/mod.rs (hunk 2 above)
Comment (teoxoy) [reply src/front/wgsl/lower/mod.rs:1152]: It was for the call a few lines below to `resolved_inner`.

**Pull request merged**